pyscriptjs/tests/integration/test_stdio_handling.py

```diff
@@ -0,0 +1,263 @@
+            </py-script>
+            """
+        )
+
+        # check line breaks at end of each input
+        assert self.page.locator("#first").inner_html() == "one.<br>two.<br>three.<br>"
+
+        # new lines are converted to line breaks
+        assert self.page.locator("#second").inner_html() == "one.<br>two.<br>three.<br>"
+
+        self.assert_no_banners()
+
+    def test_targeted_stdio_async(self):
+        # Test the behavior of stdio capture in async contexts
+        self.pyscript_run(
+            """
+            <py-script>
+                import asyncio
+                import js
+
+                async def coro(value, delay):
+                    print(value)
+                    await asyncio.sleep(delay)
+                    js.console.log(f"DONE {value}")
+            </py-script>
+
+            <div id="first"></div>
+            <py-script>
+                asyncio.ensure_future(coro("first", 1))
+            </py-script>
+
+            <div id="second"></div>
+            <py-script output="second">
+                asyncio.ensure_future(coro("second", 1))
+            </py-script>
+
+            <div id="third"></div>
+            <py-script output="third">
+                asyncio.ensure_future(coro("third", 0))
+            </py-script>
+
+            <py-script output="third">
+                asyncio.ensure_future(coro("DONE", 3))
+            </py-script>
+            """
+        )
+
+        self.wait_for_console("DONE DONE")
+
+        # py-script tags without output parameter should not send
+        # stdout to element
+        assert self.page.locator("#first").text_content() == ""
+
+        # py-script tags with output parameter not expected to send
+        # std to element in coroutine
+        assert self.page.locator("#second").text_content() == ""
+        assert self.page.locator("#third").text_content() == ""
+
+        self.assert_no_banners()
+
+    def test_targeted_stdio_interleaved(self):
+        # Test that synchronous writes to stdout are placed correctly, even
+        # While interleaved with scheduling coroutines in the same tag
+        self.pyscript_run(
+            """
+            <div id="good"></div>
+            <div id="bad"></div>
+            <py-script output="good">
+                import asyncio
+                import js
+
+                async def coro_bad(value, delay):
+                    print(value)
+                    await asyncio.sleep(delay)
+
+                print("one.")
+                asyncio.ensure_future(coro_bad("badone.", 0.1))
+                print("two.")
+                asyncio.ensure_future(coro_bad("badtwo.", 0.2))
+                print("three.")
+                asyncio.ensure_future(coro_bad("badthree.", 0))
+                asyncio.ensure_future(coro_bad("DONE", 1))
+            </py-script>
+            """
+        )
+
+        # No text should appear from coroutines
+        assert self.page.locator("#bad").text_content() == ""
```

Comment:
How would the `#bad` `div` ever be populated here?

Reply:
Ah, it wouldn't! An artifact from a refactoring, I'll delete it. Thanks!